### PR TITLE
first attempt at associations

### PR DESCRIPTION
**Pull request by bengladwell**
Here's my first attempt at getting these Active Record associations right.

One question from the start: did I handle the Hotel room_count the way you envisioned using counter_cache?

### CODE REVIEW

File: associations.rb [hunk 1]
```diff
@@ -47,25 +57,42 @@ def generate_migrations
 
 class Hotel < ActiveRecord::Base
   #insert our associations here
- 
+  has_many :rooms
+
+  def bookings
+    Booking.where(room_id: rooms.map(&:id))
```
Comment (mikegee) [inline associations.rb]: Check out ["has_many :through" associations](http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association).

**Comment by bengladwell** [comment]
Good point. Thanks.

Fixed.

**Comment by mikegee** [comment]
> Fixed

I suspect the other methods could be fixed too.

**Comment by bengladwell** [comment]
Quite right.

**Comment by bengladwell** [comment]
Thanks for your help on this @mikegee. You're the best.

**Comment by mikegee** [comment]
Aw 🥹 Miss you buddy! ❤️

**Pull request closed without merging**